### PR TITLE
[Serverless] remove random from trace-context endpoint

### PR DESCRIPTION
**Pull request by maxday**
### What does this PR do?

Use traceID and spanID from the current execution context instead of the dummy random generated values

### Motivation

make dd-trace-dotnet use the correct values

### Describe how to test/QA your changes

Unit tested

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/serverless/daemon/routes.go [hunk 1]
```diff
@@ -100,11 +99,6 @@ type TraceContext struct {
...
 	log.Debug("Hit on the serverless.TraceContext route.")
-
-	// TODO use traceID and spanID from the generated span
-	traceID := uint64(rand.Uint32())<<32 + uint64(rand.Uint32())
-	spanID := uint64(rand.Uint32())<<32 + uint64(rand.Uint32())
-
-	w.Header().Set("x-datadog-trace-id", fmt.Sprintf("%v", traceID))
-	w.Header().Set("x-datadog-span-id", fmt.Sprintf("%v", spanID))
+	w.Header().Set("x-datadog-trace-id", fmt.Sprintf("%v", invocationlifecycle.TraceID()))
+	w.Header().Set("x-datadog-span-id", fmt.Sprintf("%v", invocationlifecycle.SpanID()))
```
Comment (nhinsch) [inline pkg/serverless/daemon/routes.go:103]: I would suggest using constants for these headers

Comment (nhinsch) [reply pkg/serverless/daemon/routes.go:103]: That way we can document what they are, etc.

File: pkg/serverless/daemon/routes_test.go [hunk 2]
```diff
@@ -64,13 +65,26 @@ func TestEndInvocation(t *testing.T) {
...
+	assert.Nil(err)
+	request, err = http.NewRequest(http.MethodPost, "http://127.0.0.1:8124/trace-context", nil)
 	assert.Nil(err)
 	response, err := client.Do(request)
 	assert.Nil(err)
-	assert.NotEmpty(response.Header.Get("x-datadog-trace-id"))
-	assert.NotEmpty(response.Header.Get("x-datadog-span-id"))
+	assert.Equal("2222", fmt.Sprintf("%v", invocationlifecycle.TraceID()))
+	assert.Equal(response.Header.Get("x-datadog-trace-id"), fmt.Sprintf("%v", invocationlifecycle.TraceID()))
+	assert.Equal(response.Header.Get("x-datadog-span-id"), fmt.Sprintf("%v", invocationlifecycle.SpanID()))
```
Comment (nhinsch) [inline pkg/serverless/daemon/routes_test.go:89]: Great test! 👌

**Review by nhinsch — APPROVED** [review]
Looks good, one nit.
Of course, let's also wait until this can be merged to the working branch.

File: pkg/serverless/daemon/routes.go (hunk 1 above)
Comment (maxday) [reply pkg/serverless/daemon/routes.go]: good point! ✅

**Pull request merged**